components/view_user_group_modal/view_user_group_modal.tsx

```diff
@@ -0,0 +1,437 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {createRef, RefObject} from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import Constants, {ModalIdentifiers} from 'utils/constants';
+
+import FaSearchIcon from 'components/widgets/icons/fa_search_icon';
+import Avatar from 'components/widgets/users/avatar';
+import * as Utils from 'utils/utils.jsx';
+import LoadingScreen from 'components/loading_screen';
+import {Group} from 'mattermost-redux/types/groups';
+
+import './view_user_group_modal.scss';
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import Menu from 'components/widgets/menu/menu';
+import {ModalData} from 'types/actions';
+import AddUsersToGroupModal from 'components/add_users_to_group_modal';
+import {debounce} from 'mattermost-redux/actions/helpers';
+
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+import UpdateUserGroupModal from 'components/update_user_group_modal';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import Input from 'components/input';
+
+const USERS_PER_PAGE = 60;
+
+export type Props = {
+    onExited: () => void;
+    searchTerm: string;
+    groupId: string;
+    group: Group;
+    users: UserProfile[];
+    backButtonCallback: () => void;
+    backButtonAction: () => void;
+    currentUserId: string;
+    permissionToEditGroup: boolean;
+    permissionToJoinGroup: boolean;
+    permissionToLeaveGroup: boolean;
+    permissionToArchiveGroup: boolean;
+    actions: {
+        getGroup: (groupId: string, includeMemberCount: boolean) => Promise<{data: Group}>;
+        getUsersInGroup: (groupId: string, page: number, perPage: number) => Promise<{data: UserProfile[]}>;
+        setModalSearchTerm: (term: string) => void;
+        openModal: <P>(modalData: ModalData<P>) => void;
+        searchProfiles: (term: string, options: any) => Promise<ActionResult>;
+        removeUsersFromGroup: (groupId: string, userIds: string[]) => Promise<ActionResult>;
+        addUsersToGroup: (groupId: string, userIds: string[]) => Promise<ActionResult>;
+        archiveGroup: (groupId: string) => Promise<ActionResult>;
+    };
+}
+
+type State = {
+    page: number;
+    loading: boolean;
+    show: boolean;
+    selectedFilter: string;
+    memberCount: number;
+}
+
+export default class ViewUserGroupModal extends React.PureComponent<Props, State> {
```

Comment:
Another huge component, is there any way to subdivide this in simpler components?

Reply:
Separated this out into smaller functional components